File: src/OB.Data/Entities/Role.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace OB.Data.Entities
{
    public class Role
    {
        //--------------------------------------------------------
        [Key]
        public int Id { get; set;}

        [Required, StringLength(32)]
        public string RoleName { get; set;}

        public List<UserAccountRole> UserAccountRoles { get; set;}

        //--------------------------------------------------------
    }
}

File: _assets/output_gen/jsModels/Role.js

//---------------------------------------------------------------------------

export default class Role {
  constructor(options) {

    this.Id = null;
    this.RoleName = null;
    this.UserAccountRoles = null;

    if(options) {
      this.init(options);
    }
  }

  //---------------------------------------------------------------------------
  init(options) {

    this.Id = options.Id || this.Id;
    this.RoleName = options.RoleName || this.RoleName;
    this.UserAccountRoles = options.UserAccountRoles || this.UserAccountRoles;

  }  
  //---------------------------------------------------------------------------  
}